snapshot-labs/snapshot.js
# Refactor voting and add tests

Changes:
- Replace `jest` with `vitest`
- Add example proposals for each voting type
- Add tests for each voting type
- Add a new filter method `filterVotesWithInvalidChoice` to each voting type
- Add typescript to all voting types

Wrote tests for all voting types before making any changes to any of the classes. Also compared the results in this version with live version results and everything should be as before. Some older (~3 months and older) ranked choice voting proposals might have results change, but I checked more recent ones stay unchanged.

Additionally there was a bug on several voting types where the results would show the wrong amount for each strategy e.g: https://snapshot.org/#/ens.eth/proposal/0x3fa1a972b7d40510f5c1fb0d1c43e41960866a35db75a1541572451fa96f7eab

Before: 
![image](https://user-images.githubusercontent.com/51686767/175009977-66e5a1cb-d39c-4f9e-a2a4-2f65130ae756.png)

After: 
![image](https://user-images.githubusercontent.com/51686767/175009873-d95bd2cd-149b-4fdf-a147-956ce92c6a2f.png)
